nimare/meta/kernel.py

```diff
@@ -309,40 +311,27 @@ def __init__(self, fwhm=None, sample_size=None):
         self.sample_size = sample_size
 
     def _transform(self, mask, coordinates):
-        kernels = {}  # retain kernels in dictionary to speed things up
-        exp_ids = coordinates["id"].unique()
-
-        transformed = []
-        for i_exp, id_ in enumerate(exp_ids):
-            data = coordinates.loc[coordinates["id"] == id_]
+        ijks = coordinates[["i", "j", "k"]].values
+        exp_idx = coordinates["id"].values
 
-            ijk = np.vstack((data.i.values, data.j.values, data.k.values)).T.astype(int)
-            if self.sample_size is not None:
-                sample_size = self.sample_size
-            elif self.fwhm is None:
-                # Extract from input
-                sample_size = data.sample_size.astype(float).values[0]
-
-            if self.fwhm is not None:
-                assert np.isfinite(self.fwhm), "FWHM must be finite number"
-                if self.fwhm not in kernels.keys():
-                    _, kern = get_ale_kernel(mask, fwhm=self.fwhm)
-                    kernels[self.fwhm] = kern
-                else:
-                    kern = kernels[self.fwhm]
+        use_dict = True
```

Comment:
Thanks for including this! This is the only improvement from #709 that I wanted to get merged, so I'm glad I don't need to open a separate PR.